packages/main/src/plugin/close-behavior.ts

```diff
@@ -0,0 +1,43 @@
+/**********************************************************************
+ * Copyright (C) 2022 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+
+import { isLinux } from '../util';
+import type { ConfigurationRegistry, IConfigurationNode } from './configuration-registry';
+import type { ProviderRegistry } from './provider-registry';
+
+export class CloseBehavior {
+  constructor(private configurationRegistry: ConfigurationRegistry, private providerRegistry: ProviderRegistry) {}
+
+  async init(): Promise<void> {
+    // add configuration
+    const closeBehaviorConfigurationNode: IConfigurationNode = {
+      id: 'preferences.MinimizeToTrayOnClose',
+      title: 'Minimize To Tray On Close',
+      type: 'object',
+      properties: {
+        ['preferences.MinimizeToTrayOnClose']: {
```

Comment:
Maybe the property should be the opposite: `exitOnClose`
WDYT ?

Reply:
I'm fine with it either way. I'll change it.